### PR TITLE
Add generic delete functions in `test/util`

### PR DESCRIPTION
**Pull request by trasc**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

Add generic delete functions in `test/util`

#### Which issue(s) this PR fixes:

Fixes #

#### Special notes for your reviewer:

The relevant changeless are mainly in `test/util/util.go`

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Comment by mbobrovskyi** [comment]
/test pull-kueue-test-e2e-main-1-29

Looks like it's a common issue #2581.

File: test/util/util.go [hunk 1]
```diff
@@ -54,61 +53,31 @@ import (
...
-	return nil
+type objAsPtr[T any] interface {
+	client.Object
+	*T
 }
 
-func DeleteWorkload(ctx context.Context, c client.Client, wl *kueue.Workload) error {
-	if wl != nil {
-		if err := c.Delete(ctx, wl); err != nil && !apierrors.IsNotFound(err) {
+func DeleteObject[PtrT objAsPtr[T], T any](ctx context.Context, c client.Client, o PtrT) error {
```
Comment (alculquicondor) [inline test/util/util.go:61]:
```suggestion
func DeleteObject(ctx context.Context, c client.Client, o client.Object) error {
```

Comment (trasc) [reply test/util/util.go:61]: We'll louse the possibility to do the nil check (on the concrete object), case in which the delete will fail.  
Also the function will just be an alias to "c.Delete(ctx, o)"

Comment (trasc) [reply test/util/util.go:61]: Not really, you can detect if it's called `Delete(..., nil)` but not for `Delete(..., (*kueue.Workload)(nil))`

**Pull request merged**